vlasiator.cpp

```diff
@@ -915,12 +932,12 @@ int main(int argn,char* args[]) {
          // Copy results back from fsgrid.
          volGrid.updateGhostCells();
          technicalGrid.updateGhostCells();
-         getFieldsFromFsGrid(volGrid, BgBGrid, EGradPeGrid, technicalGrid, mpiGrid, cells);
+         getFieldsFromFsGrid(volGrid, BgBGrid, EGradPeGrid, dMomentsGrid, technicalGrid, mpiGrid, cells);
```

Comment:
Same here, update dMomentsGrid before retrieving values from it?

Reply:
Shouldn't be needed as long as the momentsgrid ghosts are updated before calculating dmoments.

Reply:
But that was missing so added update ghosts of dmoments and dmomentsdt2 just before propagatefields